src/test/java/io/debezium/connector/yugabytedb/YugabyteDBDatatypesTest.java

```diff
@@ -215,6 +269,44 @@ public void testRecordConsumption() throws Exception {
                 }).get();
     }
 
+    @Test
+    public void testRecordDeleteFieldWithYBExtractNewRecordState() throws Exception {
+        TestHelper.dropAllSchemas();
+        TestHelper.executeDDL("yugabyte_create_tables.ddl");
+        Thread.sleep(1000);
+
+        YBExtractNewRecordState<SourceRecord> transformation = new YBExtractNewRecordState<>();
+
+        Map<String, Object> configs = new HashMap<String, Object>();
+        configs.put(ExtractNewRecordStateConfigDefinition.DROP_TOMBSTONES.toString(), "false");
+        configs.put(ExtractNewRecordStateConfigDefinition.HANDLE_DELETES.toString(), "rewrite");
+        transformation.configure(configs);
+
+        String dbStreamId = TestHelper.getNewDbStreamId("yugabyte", "t1");
+        Configuration.Builder configBuilder = TestHelper.getConfigBuilder("public.t1", dbStreamId);
+        start(YugabyteDBConnector.class, configBuilder.build());
+        assertConnectorIsRunning();
+        final long rowsCount = 1;
+
+        Thread.sleep(3000);
```

Comment:
Remove the `assertConnectorIsRunning()` from line #288 and replace this `Thread.sleep(3000)` with `awaitUntilConnectorIsReady()`